cfme/tests/intelligence/test_chargeback_assignments.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+import cfme.intelligence.chargeback as cb
+import cfme.web_ui.flash as flash
+import random
+
+from utils.providers import setup_a_provider
+
+
+@pytest.fixture(scope="module")
+def provider():
+    return setup_a_provider(prov_class="infra", prov_type="virtualcenter")
+
+
+def test_assign_compute_enterprise(provider):
+
+    enterprise = cb.Assign(
+        assign_to="The Enterprise")
+    enterprise.computeassign()
+
+    flash.assert_message_match('Rate Assignments saved')
+
+
+def test_assign_compute_provider(provider):
+
+    provider_name = provider.get_yaml_data()["name"]
+    provider = cb.Assign(
+        assign_to="Selected Cloud/Infrastructure Providers",
+        selections={
+            provider_name: "Default"
+        })
+    provider.computeassign()
+
+    flash.assert_message_match('Rate Assignments saved')
+
+
+def test_assign_compute_cluster(provider):
+
+    cluster_name = provider.get_yaml_data()["clusters"]
```

Comment:
I believe it needs choice, since the clusters key is a list: `provider_name = random.choice(provider.get_yaml_data()["clusters"])`